File: solutions/lua/hamming/1/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)
    if(#a ~= #b) then    
        return -1
    end
    
    local diffCount = 0
    for i = 1, #a do
        local va = string.byte(a, i)
        local vb = string.byte(b, i)
        if(va ~= vb) then
            diffCount = diffCount + 1
        end
    end
    return diffCount
end

return Hamming

File: solutions/cpp/hamming/1/hamming.cpp
#include "hamming.h"
#include <stdexcept>
namespace hamming {
    std::size_t compute(std::string_view first, std::string_view second) {
        size_t difference = 0;
        if( first.size() != second.size())
            throw std::domain_error("");
        for(size_t i = 0; i < first.size(); ++i)
            if(first[i] != second[i])
                ++difference;
        return difference;
    }

}  // namespace hamming
